Created two anagram_checker functions to check if the two input strin…

amitsindhwani opened the pull request:
…gs are anagrams of each other.

## What changes are you trying to make? (e.g. Adding or removing code, refactoring existing code, adding reports)
Created two anagram checker functions, the first one checks if the input strings are anagrams of each other. The second function checks if the input strings are anagrams based on case sensitivity.

## What did you learn from the changes you have made?
Learnt to work on the jupyter notebook, python functions, using sets and it's methods
I was not sure if we use sorted function on a string, it would return individual letters in a sorted fashion.

## Was there another approach you were thinking about making? If so, what approach(es) were you thinking of?
One of the approach was discussed about using sorted functions in class. I used that approach initially, but tried using sets and its symmetric difference method. it worked as well! Certainly thing code is longer

## Were there any challenges? If so, what issue(s) did you face? How did you overcome it?
No major challenge, only issue was to check the syntax for using this and how it works.
Note: not sure why it is showing output [] below and how these are updated. Please let me know while reviewing this

## How were these changes tested?
tested in locally in jupyter notebook using shift+enter in the code block

## A reference to a related issue in your repository (if applicable)

## Checklist
- [Yes ] I can confirm that my changes are working as intended

pedram-aliniaye-asli commented:
You did a good job here! 
Just a note:
Your code attempts to solve the problem, but it has a fundamental issue in the logic for checking anagrams. The symmetric_difference approach is not valid for this use case, as it does not account for the frequency of characters in the strings. Here's why this logic fails:

Anagram Definition: An anagram means both strings contain exactly the same characters with the same frequencies, irrespective of order.

The symmetric difference checks for elements present in one set but not the other. It doesn't consider the frequency of characters (e.g., "aa" and "a" would incorrectly pass the check).
So please go ahead and do the modifications and then let me know. Thanks!

amitsindhwani commented:
Thanks Pedram for your feedback! I missed out on testing such scenarios when using sets. I have changed the approach by using the sorted function! I have updated the code and push the code.

pedram-aliniaye-asli commented:
you did a great job here!
Just a quick note that the branch name should be `assignment-1`, but that is not important for now.
Please go ahead and close this pull request now. Thanks!
